tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -407,6 +407,60 @@ Status FragmentConsolidator::consolidate_fragments(
         std::to_string(fragment_uris.size()) + " required fragments.");
   }
 
+  // In case we have a dense Array check that the fragments can be consolidated
+  // without data loss. More specifically, if the union of the non-empty domains
+  // of the fragments which are selected for consolidation (which is equal to
+  // the non-empty domain of the resulting consolidated fragment) overlaps with
+  // any fragment created prior to this subset, then the subset is marked as
+  // non-consolidatable. Recall that the fragment that results from
+  // consolidating a subset of fragments containing at least one dense fragment
+  // is always a dense fragment. Therefore, empty regions in the non-emtpy
+  // domain of the consolidated fragment will be filled with special values.
+  // Those values may erroneously overwrite older valid cell values
```

Comment:
```suggestion
  // Those values may erroneously overwrite older valid cell values.
  // Notes: 
  // * Consolidated dense fragments expand the non empty domain to the tile boundary.
```

Reply:
I think we also need to mention that we select the max end timestamp of all consolidated fragments as an upper bound to select fragments.